city_scrapers/spiders/stl_port_authority.py

```diff
@@ -0,0 +1,203 @@
+        rm = re.search(pattern, date)
+
+        if rm is not None:
+            day = rm.group("day")
+            time = rm.group("time")
+            dt = day + " " + time
+            end = datetime.strptime(dt.strip(), "%m/%d/%y %H:%M %p")
+            return end
+        else:
+            return None
+
+    def _parse_all_day(self, response):
+        """Parse or generate all-day status. Defaults to False."""
+        return False
+
+    def _parse_location(self, response):
+        """Parse or generate location."""
+        location = response.css("div.col-md-4 div.content-block p *::text").getall()
+        title = response.css("div.page-title-row h1::text").get().strip()
+
+        temp = []
+        for item in location:
+            item = item.replace("\n", "")
+            if item != "":
+                temp.append(item)
+        location = temp
+        i, location_index, sponsor_index = 0, 0, 0
+        while i < len(location):
+            if "location" in location[i].lower():
+                location_index = i
+            if "sponsor" in location[i].lower():
+                sponsor_index = i
+                break
+            i += 1
+
+        if "zoom" in title.lower():
+            name = "Zoom"
+            address = ""
+        elif "phone" in title.lower():
+            name = "Phone"
+            address = ""
+        elif location_index + 1 < len(location) and sponsor_index < len(location):
+            name = location[location_index + 1]
+            address = []
+            for j in range(location_index + 2, sponsor_index):
+                address.append(location[j])
+            address = (
+                " ".join(address).replace("Directions to this address", "").strip()
+            )
+        else:
+            name = ""
+            address = ""
+
+        return {
+            "address": address,
+            "name": name,
+        }
+
+    def _parse_links(self, response):
+        """Parse or generate links."""
+        agency = response.css("div.cs_control h1::text").get()
+        links = response.css("div.download li a::attr(href)").getall()
+        descriptions = response.css("div.download li a::text").getall()
+
+        patterns = {
+            "monthddcyyyy": [r"(?P<date>[A-Z][a-z]* \d{1,2}, \d{4})", "%B %d, %Y"],
+            "monthddyyyy": [r"(?P<date>[A-Z][a-z]* \d{1,2} \d{4})", "%B %d %Y"],
+        }
+
+        dt = None
+        for pattern in patterns.keys():
+            rm = re.search(patterns[pattern][0], agency)
+            if rm is not None:
+                date = rm.group("date")
+                dt = datetime.strptime(date, patterns[pattern][1])
+                break
+
+        temp_links = []
+        for link, description in zip(links, descriptions):
+            if "agenda" in description.lower() or "materials" in description.lower():
+                temp_links.append({"href": response.urljoin(link), "title": "Agenda"})
```

Comment:
The [7/9](https://www.stlouis-mo.gov/government/departments/sldc/boards/documents/port-july-9-2019.cfm) meeting's agenda link doesn't say "agenda" in the title. I think you could change it to [this](https://github.com/stl-public-meetings/city-scrapers-stl/blob/38904a3fd57549cbfba560344ceca530c4659211/city_scrapers/spiders/stl_land_reutilization.py#L216-L221).